File: 0890-lemonade-change/0890-lemonade-change.cpp
class Solution {
public:
    bool lemonadeChange(vector<int>& bills) {
        long long ch5=0,ch10=0,ch20=0;

        for(int i=0;i<bills.size();i++){
        if(bills[i] == 5){
            ch5++;
        }
        else if(bills[i] == 10){
            if(ch5<1) return false;
            ch5--; ch10++;
        }
         else if(bills[i] == 20){
         if(ch5 >=1 && ch10>=1){ch10--;ch5--;ch20++; }
            else if(ch5>=3){ch5 = ch5-3; ch20++;}
           
            else return false;
            
        }
        }
        return true;
       
    }
};

File: 0890-lemonade-change/0890-lemonade-change.java
class Solution {
    public boolean lemonadeChange(int[] bills) {
        int cnt5=0,cnt10=0,cnt20=0;
        for(int i=0;i<bills.length;i++){
             if(bills[i] == 5){
                cnt5++;
             }
             else if(bills[i] == 10){
               if(cnt5<1)return false;
               cnt5--;
               cnt10++;
             }
             else if(bills[i] == 20){
                if(cnt10 >= 1 && cnt5 >=1){
                    cnt5--;
                    cnt10--;
                    cnt20++;
                }
                else if(cnt5>=3){
                    cnt5 -= 3;
                    cnt20++;
                }
                else return false;
             }
        }
        return true;
    }
}